test/test-trace-api.js

```diff
@@ -123,6 +123,88 @@ describe('Trace Interface', function() {
       });
     });
 
+    it('should propagate rejections from a promise within a root span', function() {
+      return traceAPI.runInRootSpan({name: 'root'}, function(root) {
+        var e = new Error('Custom Message');
+        return traceAPI.runPromiseInSpan({name: 'sub'}, Promise.reject(e))
+          .then(function() {
+            assert.fail('then', 'catch', 'Should enter catch() instead of then()');
+          })
+          .catch(function(err) {
+            assert.ok(err);
+            assert.equal(err, e);
```

Comment:
It would be good to run this test asynchronously and finish in here. We want to make sure this block gets run.